### ISSUE TITLE
Does this library support collapsing and grouping?

### ISSUE THREAD

**Issue by marcofugaro** [freeman-lab/control-panel#22]
Sometimes the variables can be a lot, so grouping and collapsing of the inputs can be really useful.

Here is an example oh what I'm talking about:

http://www.curious-creature.com/pbr_sandbox/

dat.gui of course supports it and it's very useful, any plan for this library to support it?

**Comment by rreusser** [comment]
No, it doesn't. I wanted to add that, but in my opinion, I would *love* to rewrite this module with, for example, preact, so that you could make the UI nice and declarative and change it after instantiation. And so that you can compare previous and next values whenever the state changes. Long story short, it could, but there are a few hitches that prevented me from investing more time in it, even though I do love this module and use it _all the time_.

**Comment by rreusser** [comment]
Oh, and one other thing is that the values need to overlay the sliders! Pushing values into a small box makes for lots of problems in which decimal places get cut off!

Oh, and what about:

```javascript
import {Number} from 'control-panel'

controlPanel(myState, [
  <Number min=10 max=20 id="x"/>
])
```

I guess there's no reason it couldn't just allow react component instances too………

**Comment by rreusser** [comment]
One last thought is that I often just use it as a one-liner, which makes for very low mental overhead, e.g.:

```javascript
require('control-panel')([
  {type: 'number', label: '...'},
]).on('input', ...);
```

**Comment by rreusser** [comment]
Update: I've finally cleaned up my honest stab at this.

You can find a short writeup here: https://rreusser.github.io/control-panel-2/

The source lives here: https://github.com/rreusser/control-panel-2

And, naturally: https://observablehq.com/@rreusser/control-panel-2-prototype-test.
